src/pre.l

```diff
@@ -1063,6 +1065,21 @@ WSopt [ \t\r]*
                                         }
 <Command>"pragma"{B}+"once"             {
                                           yyextra->expectGuard = FALSE;
+                                          if (std::find(yyextra->pragmaList.begin(), yyextra->pragmaList.end(), yyextra->fileName.data()) != yyextra->pragmaList.end())
```

Comment:
Using `StringUnorderedSet` is much more efficient for simple "is present" type of checks (O(1) v.s. O(N), where N is the list size)